types/react-native-phone-input/index.d.ts

```diff
@@ -0,0 +1,154 @@
+
+import * as React from 'react';
+import { StyleProp, ViewStyle as ViewStyleRaw, TextStyle as TextStyleRaw, TextInput, ImageRequireSource } from 'react-native';
+
+export type ViewStyle = StyleProp<ViewStyleRaw>;
+export type TextStyle = StyleProp<TextStyleRaw>;
+
+export interface CountriesListItem {
+  name: string;
+  iso2: string;
+  dialCode: string;
+  priority: number;
+  areaCodes: ReadonlyArray<string> | null;
+}
+
+export interface PickerData {
+  key: number;
+  image: ImageRequireSource;
+  label: CountriesListItem['name'];
+  dialCode: CountriesListItem['dialCode'];
+  iso2: CountriesListItem['iso2'];
+}
+
+export interface ReactNativePhoneInputProps<TextComponentType extends React.ComponentType = typeof TextInput> {
+    /**
+     * Initial selected country
+     */
+    initialCountry?: string;
+    /**
+     * Allow user input 0 after country code
+     */
+    allowZeroAfterCountryCode?: boolean;
+    /**
+     * If true, disable all interaction of this component
+     */
+    disabled?: boolean;
+    /**
+     * Initial phone number
+     */
+    value?: string;
+    /**
+     * Custom styles to be applied if supplied
+     */
+    style?: ViewStyle;
+    /**
+     * Custom styles for flag image eg. {{width: 50, height: 30, borderWidth:0}}
+     */
+    flagStyle?: ViewStyle;
+    /**
+     * Custom styles for phone number text input eg. {{fontSize: 14}}
+     */
+    textStyle?: TextStyle;
+    /**
+     * Properties for phone number text input eg. {{placeholder: 'Telephone number'}}
+     */
+    textProps?: React.ComponentProps<TextComponentType>;
+    /**
+     * The input component to use
+     */
+    textComponent?: TextComponentType;
+    /**
+     * Distance between flag and phone number
+     */
+    offset?: number;
+    /**
+     * Set button color of country picker
+     */
+    pickerButtonColor?: string;
+    /**
+     * Set background color of country picker
+     */
+    pickerBackgroundColor?: string;
+    /**
+     * Custom styles for text in country picker eg. {{fontSize: 14}}
+     */
+    pickerItemStyle?: ViewStyle;
+    /**
+     * Cancel word
+     */
+    cancelText?: string;
+    /**
+     * Confirm word
+     */
+    confirmText?: string;
+    /**
+     * Custom styles for country picker button
+     */
+    buttonTextStyle?: TextStyle;
+    /**
+     * Function to be invoked when phone number is changed
+     */
+    onChangePhoneNumber?: (number: number) => void;
+    /**
+     * Function to be invoked when country picker is selected
+     */
+    onSelectCountry?: (iso2: string) => void;
+    /**
+     * Function to be invoked when press on flag image
+     */
+    onPressFlag?: () => void;
+    /**
+     * Custom countries list
+     */
+    countriesList?: ReadonlyArray<CountriesListItem>;
+}
+
+export default class ReactNativePhoneInput<TextComponentType extends React.ComponentType = typeof TextInput> extends React.Component<ReactNativePhoneInputProps<TextComponentType>> {
```

Comment:
There's some missing properties:
`blur`, `onPressCancel`, `onPressConfirm`

Reply:
Added in https://github.com/DefinitelyTyped/DefinitelyTyped/pull/38937/commits/66080914b81ba107d7025918c1507e3b059f28d6